hail/src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -388,9 +388,15 @@ object PruneDeadFields {
       case TableFilter(child, pred) =>
         val irDep = memoizeAndGetDep(pred, pred.typ, child.typ, memo)
         memoizeTableIR(child, unify(child.typ, requestedType, irDep), memo)
-      case TableKeyBy(child, _, isSorted) =>
+      case TableKeyBy(child, key, isSorted) =>
         val reqKey = requestedType.key
-        val childReqKey = if (isSorted) child.typ.key.take(reqKey.length) else FastIndexedSeq()
+        val childReqKey = if (isSorted)
+          child.typ.key
```

Comment:
(here, since we're asserting "trust me, I know what I'm doing" by setting `isSorted` to `true`, we can't prune any keys in the child since they might contain valuable ordering information that feed into that assertion, I think?)